src/plugins/telemetry/server/fetcher.ts

```diff
@@ -64,22 +80,74 @@ export class FetcherTask {
     this.internalRepository = new SavedObjectsClient(savedObjects.createInternalRepository());
     this.telemetryCollectionManager = telemetryCollectionManager;
 
-    this.intervalId = timer(this.initialCheckDelayMs, this.checkIntervalMs).subscribe(() =>
-      this.sendIfDue()
-    );
+    this.subscriptions.add(this.validateConnectivity());
+    this.subscriptions.add(this.startSendIfDueSubscription());
   }
 
   public stop() {
-    if (this.intervalId) {
-      this.intervalId.unsubscribe();
-    }
+    this.subscriptions.unsubscribe();
   }
 
-  private async sendIfDue() {
-    if (this.isSending) {
-      return;
-    }
+  /**
+   * Periodically validates the connectivity from the server to our remote telemetry URL.
+   * OPTIONS is less intrusive as it does not contain any payload and is used here to check if the endpoint is reachable.
+   */
+  private validateConnectivity(): Subscription {
+    return timer(this.initialCheckDelayMs, this.connectivityCheckIntervalMs)
+      .pipe(
+        // Skip any further processing if already online
+        filter(() => !this.isOnline$.value),
+        // Fetch current state and configs
+        exhaustMap(async () => await this.getCurrentConfigs()),
+        // Skip if opted-out, or should only send from the browser
+        filter(
+          ({ telemetryOptIn, telemetrySendUsageFrom }) =>
+            telemetryOptIn === true && telemetrySendUsageFrom === 'server'
+        ),
+        // Skip if already failed twice for this version
+        filter(
+          ({ failureCount, failureVersion, currentVersion }) =>
+            !(failureCount > 2 && failureVersion === currentVersion)
+        ),
```

Comment:
Probably `takeWhile` would be better here to complete the observable if we reach that condition. The performance impact of this running every 12h is not super-critical.

However, if it's failed more than 2 times, we never try again until the next version (which involves a restart), so probably closing the observable is a good thing to do.

WDYT?